libphobos/testsuite/libphobos.shared/shared.exp

```diff
@@ -39,7 +39,8 @@ proc shared_library { source destfile options } {
     global DEFAULT_DFLAGS
     global all_libraries
 
-    lappend options "additional_flags=$DEFAULT_DFLAGS -fpic -shared -shared-libphobos"
+    # Compiling with -fno-gnu-unique as tests call dlopen/dlclose multiple times on the same library.
+    lappend options "additional_flags=$DEFAULT_DFLAGS -fno-gnu-unique -fpic -shared -shared-libphobos"
```

Comment:
> I guess moving all testsuite changes to a separate commit would be nice.

OK. This is the only change to the testsuite that should be necessary. As it turns out that classinfo was never marked as weak. And by default gcc marks it as @gnu_unique_object, which means that dlclose never unloads the library.